llm/core/types.py

```
@@ -0,0 +1,174 @@
+    runtime: RuntimeConfig
+    embed_tokens: torch.Tensor
+    final_norm_weight: torch.Tensor
+    lm_head: torch.Tensor
+    layers: list[LayerWeights]
+
+
+@dataclass
+class ModelRecord:
+    config: ModelConfig
+    runtime: RuntimeConfig
+    tokenizer: "TokenizerAdapter"
```

Ruff (0.15.10)

[error] Undefined name `TokenizerAdapter`

(F821)